### PR TITLE
show missed profit

### LINKED ISSUE #293: Sum up and display missed estimated profit
It is not apparent how much better the premium version is compared to the free one.
To quantify this there should be a summary on `/flipper` that sums up the following:
* total flips
* missed flips count (SOLD)
* missed estimated profit
* estimated profit of copied auctions

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #293

### CODE REVIEW

**Comment by Flou21** [comment]
created temp environment: https://temp-env-sky-339.flou21.de

File: src/components/Flipper/Flipper.tsx [hunk 1]
```diff
@@ -236,13 +269,38 @@ function Flipper() {
...
+                                            <li>Profit of copied flips: {numberWithThousandsSeperators(missedInfo.estimatedProfitCopiedAuctions)} Coins</li>
+                                        </ul>
+                                    </fieldset>
+                                    <fieldset>
+                                        <legend>You dont have Premium?</legend>
+                                        <ul>
+                                            <li>
+                                                <span style={{marginRight: "10px"}}>Missed Profit: {numberWithThousandsSeperators(missedInfo.missedEstimatedProfit)} Coins</span>
+                                                <Tooltip type="hover" content={<HelpIcon />}
+                                                    tooltipContent={<span>The premium flipper is not delayed. Therefor you get access to the most profitable flips. Currently really good flips are already sold.</span>} />
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:296]:
```suggestion
                                                    tooltipContent={<span>The premium flipper is not delayed. Therefor you would get access to the most profitable flips and not miss out on this amount of profit.</span>} />
```

**Review by Ekwav — CHANGES_REQUESTED** [review]
Please move this header below the flipper-card into its own card.
Make it collapsed by default and display only 
"Missed Profit: XY " as the title.
In its current form its way to annoying especially if you click on a flip and it scrolls up.

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

File: src/components/Flipper/Flipper.tsx [hunk 2]
```diff
@@ -302,6 +334,40 @@ function Flipper() {
...
+                                            <li>Profit of copied flips: {numberWithThousandsSeperators(missedInfo.estimatedProfitCopiedAuctions)} Coins</li>
+                                        </ul>
+                                    </fieldset>
+                                    <fieldset>
+                                        <legend>You dont have Premium?</legend>
+                                        <ul>
+                                            <li>
+                                                <span style={{ marginRight: "10px" }}>Missed Profit: {numberWithThousandsSeperators(missedInfo.missedEstimatedProfit)} Coins</span>
+                                                <Tooltip type="hover" content={<HelpIcon />}
+                                                    tooltipContent={<span>The premium flipper is not delayed. Therefor you would get access to the most profitable flips and not miss out on this amount of profit.</span>} />
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:360]:
```suggestion
                                                    tooltipContent={<span>This is the sum of the field 'Estimated profit' of the flips that were already sold when you received them. It represents the extra coins you could earn if you purchased our premium plan.</span>} />
```

File: src/components/Flipper/Flipper.tsx [hunk 3]
```diff
@@ -302,6 +334,40 @@ function Flipper() {
...
+                                <div style={{ display: "flex", alignContent: "center", justifyContent: "space-around" }}>
+                                    <fieldset>
+                                        <legend>You got:</legend>
+                                        <ul>
+                                            <li>Total found flips: {numberWithThousandsSeperators(missedInfo.totalFlips)}</li>
+                                            <li>Profit of copied flips: {numberWithThousandsSeperators(missedInfo.estimatedProfitCopiedAuctions)} Coins</li>
+                                        </ul>
+                                    </fieldset>
+                                    <fieldset>
+                                        <legend>You dont have Premium?</legend>
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:355]:
```suggestion
                                        <legend>You don't have Premium</legend>
```

File: src/components/Flipper/Flipper.tsx [hunk 4]
```diff
@@ -302,6 +334,40 @@ function Flipper() {
...
+                        <Card>
+                            <Card.Header>
+                                <Card.Title>Missed profit</Card.Title>
+                            </Card.Header>
+                            <Card.Body>
+                                <div style={{ display: "flex", alignContent: "center", justifyContent: "space-around" }}>
+                                    <fieldset>
+                                        <legend>You got:</legend>
+                                        <ul>
+                                            <li>Total found flips: {numberWithThousandsSeperators(missedInfo.totalFlips)}</li>
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:350]:
```suggestion
                                            <li>Total flips received: {numberWithThousandsSeperators(missedInfo.totalFlips)}</li>
```

File: src/components/Flipper/Flipper.tsx [hunk 5]
```diff
@@ -302,6 +334,40 @@ function Flipper() {
                     </div> : ""
             }
 
+            {
+                !isLoading && !hasPremium ?
+                    <div>
+                        <hr />
+                        <Card>
+                            <Card.Header>
+                                <Card.Title>Missed profit</Card.Title>
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:343]:
```suggestion
                                <Card.Title>Flipper summary</Card.Title>
```

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

File: src/components/Flipper/Flipper.tsx [hunk 6]
```diff
@@ -302,20 +334,59 @@ function Flipper() {
...
+                                            <ul>
+                                                <li>Total flips received: {numberWithThousandsSeperators(missedInfo.totalFlips)}</li>
+                                                <li>Profit of copied flips: {numberWithThousandsSeperators(missedInfo.estimatedProfitCopiedAuctions)} Coins</li>
+                                            </ul>
+                                        </Card.Body>
+                                    </Card>
+                                    <Card className="flipper-summary-card">
+                                        <Card.Header>
+                                            <Card.Title>
+                                                You don't have Premium?
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:363]:
```suggestion
                                                You don't have Premium
```

File: src/components/Flipper/Flipper.tsx [hunk 7]
```diff
@@ -302,20 +334,59 @@ function Flipper() {
...
+                                                <li>
+                                                    <span style={{ marginRight: "10px" }}>Missed Profit: {numberWithThousandsSeperators(missedInfo.missedEstimatedProfit)} Coins</span>
+                                                    <Tooltip type="hover" content={<HelpIcon />}
+                                                        tooltipContent={<span>This is the sum of the field 'Estimated profit' of the flips that were already sold when you received them. It represents the extra coins you could earn if you purchased our premium plan</span>} />
+                                                </li>
+                                                <li>Missed Flips: {numberWithThousandsSeperators(missedInfo.missedFlipsCount)}</li>
+                                            </ul>
+                                        </Card.Body>
+                                    </Card>
+                                    <Card className="flipper-summary-card">
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:377]: this card should grow if there is space

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

File: src/components/Flipper/Flipper.tsx [hunk 8]
```diff
@@ -335,7 +335,7 @@ function Flipper() {
             }
 
             {
-                !isLoading && isLoggedIn && !hasPremium ?
+                !isLoading && isLoggedIn && hasPremium ?
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:338]: I am pretty sure that you want to revert this

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-339.flou21.de

**Review by Ekwav — APPROVED** [review]

**Pull request merged**